### PR TITLE
VSTS-136 Automatically pass properties needed for branch/PR analysis

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: common/ts/prepare-task.ts [hunk 1]
```diff
@@ -8,6 +9,9 @@ export default async function prepareTask(endpoint: Endpoint, rootPath: string)
   const scanner = Scanner.getPrepareScanner(rootPath, scannerMode);
 
   const props: { [key: string]: string } = {};
+
+  await populateBranchAndPrProps(endpoint, props);
```
[the remark this answers is not in the document]
Comment (gregaubert) [reply common/ts/prepare-task.ts]: Let's also put a debug log juste after the populate : 
```TS
if (endpoint.type === EndpointType.SonarCloud) {
  await populateBranchAndPrProps(endpoint, props);
  tl.debug(`[SQ] Branch and PR parameters: ${JSON.stringify(props)}`);
}
```

File: common/ts/package.json [hunk 2]
```diff
@@ -7,7 +7,8 @@
   "dependencies": {
     "fs-extra": "5.0.0",
     "request": "2.83.0",
-    "vsts-task-lib": "2.1.0"
+    "vsts-task-lib": "2.1.0",
+    "vso-node-api": "6.2.8-preview"
```
[the remark this answers is not in the document]
Comment (henryju) [reply common/ts/package.json]: No reason, I'll update

File: common/ts/publish-task.ts [hunk 3]
```diff
@@ -39,14 +37,15 @@ export default async function publishTask(endpointType: EndpointType) {
     if (e instanceof TimeOutReachedError) {
       tl.warning(
         `Task '${
-          task.analysisId
+          taskReport.ceTaskId
```
Comment (gregaubert) [inline common/ts/publish-task.ts]: We don't have the `task` if a timeout error is thrown

Comment (henryju) [reply common/ts/publish-task.ts]: Good catch!

**Comment by lpmatos** [comment]
This plugin support when i don't want a branch analysis?

**Comment by mickael-caro-sonarsource** [comment]
Hi @lpmatos 

What are you needs exaclty ?

**Comment by lpmatos** [comment]
Hi @mickael-caro-sonarsource!

I am building a pipeline in Azure Devops and I have the sonar job. In my infrastructure, sonarqube runs on version 8.5.1 of the community on an AKS cluster. 

To use this integration I install this plug-in in my organization through the azure marketplace, however, whenever the job is executed, it fails in the following message:

image

I would like that this was optional or that had a documentation with the step by step of this configuration.

**Pull request merged**